test/rules/no-shadowed-variable/default/ambient.d.ts.lint

```diff
@@ -0,0 +1,7 @@
+// no errors in declaration files
```

Comment:
i'm not sure about this -- why should we skip declaration files entirely? Is the idea that declaration files have to adhere to all the idiosyncrasies of their corresponding JS source and therefore can't help it if it has shadowed names, so it's impossible to avoid the shadowing unless the JS is fixed as well?

Reply:
Exactly. We skip declartation files for the same reason we ignore module augmentations: they model something that already exists. Like you said:

> declaration files have to adhere to all the idiosyncrasies of their corresponding JS source and therefore can't help it if it has shadowed names, so it's impossible to avoid the shadowing unless the JS is fixed as well?

Reply:
cool, we're on the same page.